### PR TITLE
Code-review fixes for rotation (#41 and #42)

### PR DESCRIPTION
**Pull request by ashes999**
`(x, y)` coordinates are optional in `rotate`
Verified translation for Matrix is negated correctly

Please review this @ibilon . The translation looks negated, I think that's because the Haxe matrix class inverts it. It is actually the correct formula (negating it gives the wrong rotation).

### CODE REVIEW

**Comment by ashes999** [comment]
Just a reminder that I'm still waiting for this @ibilon

**Comment by ibilon** [comment]
Oh sorry, was on holiday when you posted and I missed it,
will review soon.

**Comment by ibilon** [comment]
I'd change the regex to `~/rotate\(([0-9\.]+)(\s+([0-9\.]+)\s+([0-9\.]+))?\)/`
according to the doc there's no comma,
you can't just pass "angle x", it's either "angle" or "angle x y",
also you need at least a space between them.

**Comment by ashes999** [comment]
Here's my dilemma. TLDR: compatibility or strict adherence to specifications?

You're right that the SVG docs don't specify a comma here.  However, I generated my test SVGs with `svg-edit`, which used commas in the generated SVG.

I would prefer an SVG library that "just works" with SVGs from anywhere, even if they're not (strictly) adherent to specifications. This will close the door on a whole class of bugs people may open in the future about "well I used X program for my SVG and it doesn't show up properly."

On the other hand, maybe looser validation will bite us. We can't possibly support every (broken?) SVG tool out there. So we should stick to the specs.

What are your opinions on this @ibilon  and @singmajesty ?

**Comment by jgranick** [comment]
I think the motivation is to load our SVG files, not to get tripped up on specifics :wink:

**Comment by ashes999** [comment]
So can I merge?

**Comment by ibilon** [comment]
Yeah I guess since the input is user provided svg it makes sense to be lenient about standards.

@ashes999 still need a couple of changes to the regex, the first `\s*` should be `\s+`,
maybe add a `\s*` before the comma, and change the `,?` with a `[, ]`.
So: `~/rotate\(([0-9\.]+)(\s+([0-9\.]+)\s*[, ]\s*([0-9\.]+))?\)/`

File: format/svg/SVGData.hx [hunk 1]
```diff
@@ -33,7 +33,7 @@ class SVGData extends Group {
 	private static var mTranslateMatch = ~/translate\((.*)[, ](.*)\)/;
 	private static var mScaleMatch = ~/scale\((.*)\)/;
 	private static var mMatrixMatch = ~/matrix\((.*?)[, ]+(.*?)[, ]+(.*?)[, ]+(.*?)[, ]+(.*?)[, ]+(.*?)\)/;
-	private static var mRotationMatch = ~/rotate\(([0-9\.]+) ([0-9\.]+),([0-9\.]+)\)/;
+	private static var mRotationMatch = ~/rotate\(([0-9\.]+)(\s+([0-9\.]+),?\s+([0-9\.]+))?\)/;
 	private static var mURLMatch = ~/url\(#(.*)\)/;
```
Comment (ibilon) [inline format/svg/SVGData.hx]: now this forces a space after the comma, which the test svg doesn't have

Comment (ashes999) [inline format/svg/SVGData.hx]: I committed this change superficially.

**Comment by ashes999** [comment]
This should fix all the remaining issues. The tests also pass now.

**Comment by ibilon** [comment]
:+1:

**Pull request merged**